challenge_3/go/makernaren/max_occurance.go

```diff
@@ -0,0 +1,38 @@
+// Take an array of mixed data types and find the element that occurs
+// most number of times.
+package main
+
+import (
+	"fmt"
+)
+
+// Finds the maximum occurance, takes in interface as input and output params
+//as elements in the given array may be of any data type
+func FindMajority(input []interface{}) (result interface{}) {
```

Comment:
A slice of empty interfaces is unwieldy. Sure, it's Go's iffy "generics", but if you have a typed slice, it has to be copied into this new generic slice before being passed in. You would likely end up just writing a FindMajorityTYPE for each type you wanted this functionality for or you'd create a new interface that better supported what this needs to work.

For this problem, it's ok, as you create the slice of empty interfaces as input and it's not used for anything else.